src/SourceBuild/content/test/Microsoft.DotNet.Tests/PackageSourceMappingsTests.cs

```diff
@@ -0,0 +1,242 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Xml.Linq;
+using Microsoft.DotNet.UnifiedBuild.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.Tests
+{
+    [Trait("Category", "PackageSourceMappings")]
+    public class PackageSourceMappingsTests
+    {
+        private static readonly PackageSourceMappingsSetup TestSetup = PackageSourceMappingsSetup.Instance;
+        private ITestOutputHelper OutputHelper { get; }
+
+        public PackageSourceMappingsTests(ITestOutputHelper outputHelper)
+        {
+            OutputHelper = outputHelper;
+        }
+
+        // Unified build - with existing mappings - online
+        [Fact]
+        public void UnifiedBuildWithMappings()
+        {
+            string[] sources = ["source-built-arcade", "source-built-runtime", "net-sdk-supporting-feed"];
+            RunTest("ub-mappings.config", true, sources, customSources: ["net-sdk-supporting-feed"], sourceBuild: false);
+        }
+
+        // Unified build - with mappings - online - no local sources
+        [Fact]
+        public void UnifiedBuildWithMappingsNoLocalSources()
+        {
+            string[] sources = ["net-sdk-supporting-feed"];
+            RunTest("ub-mappings-nolocal.config", true, sources, customSources: ["net-sdk-supporting-feed"], sourceBuild: false);
+        }
+
+        // Unified build - no mappings - online
+        [Fact]
+        public void UnifiedBuildNoMappings()
+        {
+            string[] sources = ["source-built-arcade", "source-built-runtime", "net-sdk-supporting-feed"];
+            RunTest("ub.config", true, sources, customSources: ["net-sdk-supporting-feed"], sourceBuild: false);
+        }
+
+        // Source build tests - with and without mappings - online and offline
+        [Theory]
+        [InlineData("sb-online.config", true)]
```

Comment:
I personally think it would help to name the NuGet.configs with a consistent explicit naming pattern.  Specifically I think it would help to call our noMappings in the files.  This is similar in nature how you have both online/offline in the name versus implying a default if not specified.

Reply:
Makes sense - will update.

Reply:
Fixed with https://github.com/dotnet/sdk/pull/44649/commits/f13f4f1cfb5a7adec023571eaf49ddcc8a051648